File: backend-bot/src/main/java/cc/jktu/apps/bot/controller/BotController.java
package cc.jktu.apps.bot.controller;

import cc.jktu.apps.bot.service.BotService;
import cc.jktu.apps.common.CommonResp;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/bot")
public class BotController {

    private final BotService botService;

    public BotController(BotService botService) {
        this.botService = botService;
    }

    @PostMapping("/reLogin")
    public CommonResp<Void> reLogin() {
        botService.getBot().login();
        return CommonResp.empty();
    }

}

File: backend-bot/src/main/kotlin/cc/jktu/apps/bot/controller/BotController.kt
package cc.jktu.apps.bot.controller

import cc.jktu.apps.bot.service.BotService
import cc.jktu.apps.common.util.CommonResp
import cc.jktu.apps.common.util.emptyResp
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import kotlinx.coroutines.runBlocking
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/bot")
@Tag(name = "qq机器人接口")
class BotController(private val botService: BotService) {

    @PostMapping("/reLogin")
    @Operation(summary = "重新登陆")
    fun reLogin(): CommonResp<Nothing?> {
        runBlocking { botService.bot.login() }
        return emptyResp()
    }
}
